sqlmesh/utils/jinja.py

```diff
@@ -228,9 +228,24 @@ def extract_macro_references_and_variables(
                         )
                     )
 
+        def is_variable_node(n: nodes.Node) -> bool:
+            return (
+                isinstance(n, nodes.Call)
+                and isinstance(n.node, nodes.Name)
+                and n.node.name in (c.VAR, c.BLUEPRINT_VAR)
+            )
+
         for call_name, node in extract_call_names(jinja_str):
             if call_name[0] in (c.VAR, c.BLUEPRINT_VAR):
-                assert isinstance(node, nodes.Call)
+                if not is_variable_node(node):
+                    # Find the variable node which could be nested
+                    for n in node.find_all(nodes.Call):
+                        if is_variable_node(n):
+                            node = n
+                            break
+                    else:
+                        raise ValueError(f"Could not find variable name in {jinja_str}")
```

Comment:
Should we just push that logic into the helper and call it like `extract_variable_node` or something?